sdk/servicebus/Azure.Messaging.ServiceBus/src/Processor/ServiceBusProcessor.cs

```diff
@@ -349,18 +370,29 @@ public virtual async Task CloseAsync(CancellationToken cancellationToken = defau
             cancellationToken.ThrowIfCancellationRequested<TaskCanceledException>();
             if (ActiveReceiveTask == null && MessageHandlerSemaphore == null)
             {
-                MessageHandlerSemaphore = new SemaphoreSlim(
-                    MaxConcurrentCalls,
-                    MaxConcurrentCalls);
+                ServiceBusEventSource.Log.StartProcessingStart(Identifier);
                 await ProcessingStartStopSemaphore.WaitAsync(cancellationToken).ConfigureAwait(false);
+                if (_processMessage == null)
+                {
+                    throw new InvalidOperationException(string.Format(CultureInfo.CurrentCulture, Resources1.CannotStartEventProcessorWithoutHandler, nameof(ProcessMessageAsync)));
```

Comment:
`Resources1`? 😄

Reply:
Hehe, yeah we have an issue to go and fix this - https://github.com/Azure/azure-sdk-for-net/issues/9918